### ISSUE TITLE
Calling `$(...)` results in a segmentation fault

### ISSUE THREAD

**Issue by 9Morello** [oven-sh/bun#8728]
### What version of Bun is running?

1.0.26+c75e768a6

### What platform is your computer?

Ubuntu 22.04

### What steps can reproduce the bug?

1. Run the following code:
```js
import { $ } from "bun";

await $(`echo "Hello World"`);
```

### What is the expected behavior?

Either the code should print "Hello World" in the terminal, or warn about a syntax error if the built-in shell isn't meant to be used this way.

### What do you see instead?

`Segmentation fault (core dumped)`

### Additional information

I don't know if this happens on macOS as well.

**Comment by Electroid** [comment]
The right usage is:
```js
await $`echo "Hello World"`
```

But obviously this shouldn't segfault, we'll fix this. cc @zackradisic

**Comment by zackradisic** [comment]
Fixed in #8846
